Address comments and NITS from SEC and INT area (#183)
mlswg/mls-architecture: draft-ietf-mls-architecture.md

managed in GitHub, but any substantive change should be discussed on
the MLS mailing list.

End-to-end security is a used in the vast majority of instant messaging systems,
and also deployed in systems for other purposes such as calling and conferencing.

> In this context, "end-to-end" captures

Reviewer: This is where we should reference https://datatracker.ietf.org/doc/draft-knodel-e2ee-definition/ instead of having our own definition

Author: This paragraph is not a definition. It is a statement of adoption.  The only aspect of e2e we specifically call out is that the operator of the messaging system can be considered an attacker. 

As for including this reference anyway, looking at the status, this has several Not Ready telechat ballot positions, so I am inclined to leave it out.

* MLS clients connected to a peer-to-peer network could instantiate a
  decentralized DS by transmitting MLS messages over that network.

* In an MLS group using a Public Key Infrastructure (PKI) for authentication,
  the AS would comprise the certificate issuance and validation processes,
  both of which involve logic inside MLS clients as well as various

>   existing PKI roles (ex: Certificate Authorities).

Reviewer 2: ```suggestion
  existing PKI roles (ex: Certification Authorities).
```

Reviewer 3: Please change "ex:" to "e.g.,"

Roughly speaking, "deniability" is the opposite of "non-repudiation", i.e., the
property that it is impossible to prove to a third party that a message was sent
by a given sender. Deniability is not currently provided by the MLS protocol, but the
protocol avoids constraints that would make it impossible to add deniability

> properties (via extensions) in the future. Defining the specific requirements

Reviewer 4: Recommend striking "in the future". This seems to imply that the potential for deniability is linked to follow-on iterations of MLS and cannot be achieved in its current state (i.e. based on implementation decisions). Note that if kept, "in the future" does not imply only to the current time but to the full lifetime of the standard.

> properties (via extensions) in the future. Defining the specific requirements

Author: ```suggestion
properties (via extensions). Defining the specific requirements
```

- The attacker can publish or distribute credentials

> It is possible for a system to have a centralized server generate signature key

Reviewer 5: ```suggestion
It is possible for a system to have a central server generate signature key
```

Author: Keeping centralized for consistency (appears twice more in the same paragraph)

Details about how to verify the identity of a client depend on the MLS
Credential type used. For example, cryptographic verification of credentials can

> be largely performed autonomously (e.g. without additional network traffic) by

Reviewer 5: ```suggestion
be largely performed automatically by
```
Trying to convey that X.509 is not manual, not that it doesn't require network (even though it may not)

Author: I was trying to convey that it doesn't require any additional network traffic, as this was the genesis of the original comment/question.

Reviewer 5: X.509 verification does require network traffic though, to check revocation. The "In contrast" below is talking about out-of-band / manual verification

> be largely performed autonomously (e.g. without additional network traffic) by

Author: ```suggestion
be largely performed autonomously (e.g. without user interaction) by
```
